Framework/DataHandling/src/RotateSource.cpp

```diff
@@ -0,0 +1,137 @@
+#include "MantidDataHandling/RotateSource.h"
+#include "MantidDataObjects/Workspace2D.h"
+#include "MantidDataObjects/PeaksWorkspace.h"
+#include "MantidGeometry/Instrument/ReferenceFrame.h"
+#include "MantidKernel/Quat.h"
+#include "MantidKernel/V3D.h"
+
+namespace Mantid {
+namespace DataHandling {
+
+using Mantid::Kernel::Direction;
+using Mantid::Kernel::Quat;
+using Mantid::Kernel::V3D;
+using Mantid::API::WorkspaceProperty;
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(RotateSource)
+
+using namespace Geometry;
+using namespace API;
+
+//----------------------------------------------------------------------------------------------
+/** Initialize the algorithm's properties.
+ */
+void RotateSource::init() {
+
+  // When used as a Child Algorithm the workspace name is not used - hence the
+  // "Anonymous" to satisfy the validator
+  declareProperty(new WorkspaceProperty<Workspace>("Workspace", "Anonymous",
```

Comment:
I think the input workspace should be a type of MatrixWorkspace? I don't think we'd want to allow rotation of a TableWorkspace for instance.